source/common/buffer/buffer_impl.h

```diff
@@ -419,9 +422,10 @@ class Slice {
   /** Account associated with this slice. This may be null. When
    * coalescing with another slice, we do not transfer over their account. */
   BufferMemoryAccountSharedPtr account_;
-};
 
-class OwnedImpl;
+private:
+  friend OwnedImpl;
```

Comment:
Do we need to make OwnedImpl a friend? I think we can get away with using the public interface via reservation, commit to the slice in https://github.com/envoyproxy/envoy/pull/19115/files#diff-0fdef7bd42dfa48ed5a015ddb85be158a514d61a82a5a17c1cfa34c6d744e5f8R604-R610

Reply:
Can this be removed now that we use the public interface? Thanks!